### DISCUSSION TITLE
Updating funds

### DISCUSSION THREAD

**Discussion by thal-sh** [pocketbase/pocketbase#4734] [Q&A]
I'm making an app where u have funds (money) and if you buy, your amount should be deducted i wanna ask, would it go out of sync if i use a base collection

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what do you mean by _"go out of sync if i use a base collection"_.

A Base collection stores its data as plain SQLite table under the same name as the collection and each collection schema field is a column in that specific table.

If by "out of sync" you mean sending 2 separate requests (one to retrieve the old value and another to update it with a new one) or even just changing the value with the `+/-` modifiers then there is no guarantee that while your request is being processed that someone else will not have modified the fund value already.

If you want your operation to be atomic and to avoid race conditions you'll have to execute the read and write operation in a transaction. This currently can be done only programmatically using [Go](https://pocketbase.io/docs/go-records/#transaction) or the [JS pb_hooks](https://pocketbase.io/docs/js-records/#transaction).

Support for batch and transactional operations (currently write only) via the Web APIs is already implemented as part of the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) but it will take a while before it could be released.
